packages/xo-server/src/xapi/mixins/vm.mjs

```diff
@@ -491,7 +492,12 @@ const methods = {
     hasVendorDevice: true,
 
     expNestedHvm: {
-      set: (expNestedHvm, vm) => vm.update_platform('exp-nested-hvm', expNestedHvm ? 'true' : null),
+      set: (expNestedHvm, vm) => {
+        const platformVersion = semver.satisfies(vm.hardware_platform_version, '>=3.4')
+          ? 'nested-virt'
+          : 'exp-nested-hvm'
+        vm.update_platform(platformVersion, expNestedHvm ? 'true' : null)
+      },
```

Comment:
Same here, I found that weird having a `set expNestedHvm` that in reality will updated `nested-virt`

Reply:
would it be ok to update both ? 
also this could help handling VM that are migrated from 8.2 to 8.3 hosts